.github/workflows/checks.yml

```diff
@@ -1,53 +1,36 @@
 name: Checks
 
+# Run on push to main and any pull request.
+# Allow manual execution of the workflow.
 on:
   push:
     branches:
       - main
   pull_request:
+  workflow_dispatch:
 
 permissions:
+  checks: write # Allow write access to checks to allow annotation of code in the PR.
   contents: read
   pull-requests: read
 
 jobs:
-
   checks:
-    name: Check Process
     runs-on: ubuntu-latest
-    env:
-      GO_VERSION: oldstable
-      GOLANGCI_LINT_VERSION: v1.56.2
-      CGO_ENABLED: 0
```

Comment:
It is better for the maintenance to have all the configurable elements in one place.

Reply:
Why use oldstable for these checks rather than stable?

Reply:
Because the previous CI was doing that.
You can use `stable`, it's not a problem for me, but I still recommend centralizing the variables, it helps for the maintenance and to diagnose CI problems.